bindings/python/edlib.pyx

```diff
@@ -152,3 +154,98 @@ def align(query, target, mode="NW", task="distance", k=-1, additionalEqualities=
+    """ Align query with reference using edit distance, and output alignments in NICE format
+    @param {str or bytes or iterable of hashable objects} query, combined with reference must have no more
+           than 256 unique values
+    @param {str or bytes or iterable of hashable objects} reference, combined with query must have no more
+           than 256 unique values
+    @param {string} mode  Optional. Alignment method do be used. Possible values are:
+            - 'NW' for global (default)
+            - 'HW' for infix
+            - 'SHW' for prefix.
+    @param {int} k  Optional. Max edit distance to search for - the lower this value,
+            the faster is calculation. Set to -1 (default) to have no limit on edit distance.
+    @param {list} additionalEqualities  Optional.
+            List of pairs of characters or hashable objects, where each pair defines two values as equal.
+            This way you can extend edlib's definition of equality (which is that each character is equal only
+            to itself).
+            This can be useful e.g. when you want edlib to be case insensitive, or if you want certain
+            characters to act as a wildcards.
+            Set to None (default) if you do not want to extend edlib's default equality definition.
+    @return Dictionary with following fields:
+            {string} cigar  Cigar is a standard format for alignment path.
+                Here we are using extended cigar format, which uses following symbols:
+                Match: '=', Insertion to reference: 'I', Deletion from reference: 'D', Mismatch: 'X'.
+                e.g. cigar of "5=1X1=1I" means "5 matches, 1 mismatch, 1 match, 1 insertion (to reference)".
+            {string} NICE  NICE is the indented pairwise alignment. 
+            {string} query_aln  query_aln is the aligned query sequence from the input 'query'
+            {string} match_aln  match_aln is the pairwise alignment from the parsed cigar
+                The symbol '|' represents a read match '='
+                The symbol '.' represents a read mismtach 'X'
+                The space ' ' represents an insertion 'I' or deletion 'D'
+            {string} ref_aln  ref_aln is the aligned reference sequence from the input 'reference'                
+    """
+
+    ## 'path' - find edit distance, start and end locations and alignment path. (Default)
+
+    result = align(query=query, target=reference, mode=mode, task="path", k=k, additionalEqualities=additionalEqualities)
+
+    ## access locations, a list of a set [(start, end)], from yech1990
+    ## ref_pos = locations[0][0]
+    ref_pos = result["locations"][0][0]
+    if ref_pos == None:
+        ref_pos = 0
+    query_pos = 0 ## 0-indexed
+    ref_aln = match_aln = query_aln = ""
+
+    cigar = result["cigar"]
+
+    ## cigar parsing, from yech1990
+    for step, code in re.findall("(\d+)(\D)", cigar):
```

Comment:
step, code -> those names don't seem to be exactly on point. If we have 3X, I would say 3 is number of occurrences (what would step mean?) and X is alignment operation / symbol (code is not so terrible, but still not completely on spot).

Reply:
I took the above variables directly from @yech1990

How about `occurrence, symbol in re.findall("(\d+)(\D)", cigar):`? Whatever you choose here will be suitable for me

Reply:
Ah yes I assumed you took them from there, but now they are under scrutiny :D.
I don't care about exact name, as long as it describes what it is.
Let's see: The first thing I would say is "Number of occurrences of the alignment operation". "occurence" is in right direction, but a lot of info is still missing, it does not even sound like a number, and we are dealing with a number here, so I think we can do better. "num_occurences" for example sounds better to me. We could go further with "num_alignment_operation_occurences" but that is really long and since we will describe the second variable with nice name, I think it will be redundant. So "num_occurences" sounds best.
The second thing I would say is "Cigar symbol/code that represent an alignment operation". Since we already know we are in cigar, most information is probably delivered with "alignment_operation", so I would probably go for that.

Reply:
I like the variable names, yes!

`for num_occurrences, alignment_operation in re.findall("(\d+)(\D)", cigar):
`

And a short comment will clarify any possible ambiguity. This looks good!

Reply:
Great :)!
As mentioned in "Clean code", comments should be used only when we failed to express what we wanted through the code :D, but I think in this case I agree comment can bring extra clarity and trying to express that through code just does not feel like reasonable effort.